pdr_utils/contract.py

```diff
@@ -0,0 +1,331 @@
+            return nom/denom
+        except Exception as e:
+            print("Failed to call getAggPredval")
+            print(e)
+            return None
+
+
+    
+    def payout(self,slot):
+        """ Claims the payout for a slot"""
+        gasPrice = w3.eth.gas_price
+        try:
+            tx = self.contract_instance.functions.payout(slot,owner).transact({"from":owner,"gasPrice":gasPrice})
+            print(f"Submitted payout, txhash: {tx.hex()}")
+            receipt = w3.eth.wait_for_transaction_receipt(tx)
+            return receipt
+        except Exception as e:
+            print(e)
+            return None
+
+    def soonest_block_to_predict(self,block):
+        return self.contract_instance.functions.soonestBlockToPredict(block).call()
+    
+    def submit_prediction(self,predicted_value,stake_amount,prediction_block):
+        """ Sumbits a prediction"""
+        stake_token = Token(self.contract_instance.functions.stakeToken().call())
+        # TO DO - check allowence
+        amount_wei = w3.to_wei(str(stake_amount),'ether')
+        stake_token.approve(self.contract_address,amount_wei)
+        gasPrice = w3.eth.gas_price
+        try:
+            tx = self.contract_instance.functions.submitPredval(predicted_value,amount_wei,prediction_block).transact({"from":owner,"gasPrice":gasPrice})
+            print(f"Submitted prediction, txhash: {tx.hex()}")
+            receipt = w3.eth.wait_for_transaction_receipt(tx)
+            return receipt
+        except Exception as e:
+            print(e)
+            return None
+    
+    def get_trueValSubmitTimeoutBlock(self):
+        return self.contract_instance.functions.trueValSubmitTimeoutBlock().call()
+    
+    def get_prediction(self,slot):
+        return self.contract_instance.functions.getPrediction(slot).call({"from":owner})
+
+    def submit_trueval(self,true_val,block,float_value,cancel_round):
+        gasPrice = w3.eth.gas_price
+        try:
+            fl_value=w3.to_wei(str(float_value),'ether')
+            tx = self.contract_instance.functions.submitTrueVal(block,true_val,fl_value,cancel_round).transact({"from":owner,"gasPrice":gasPrice})
+            print(f"Submitted trueval, txhash: {tx.hex()}")
+            receipt = w3.eth.wait_for_transaction_receipt(tx)
+            return receipt
+        except Exception as e:
+            print(e)
+            return None
+    
+    def redeem_unused_slot_revenue(self,block):
+        gasPrice = w3.eth.gas_price
+        try:
+            tx = self.contract_instance.functions.redeemUnusedSlotRevenue(block).transact({"from":owner,"gasPrice":gasPrice})
+            print(f"redeem_unused_slot_revenue tx: {tx.hex()}")
+            receipt = w3.eth.wait_for_transaction_receipt(tx)
+            return receipt
+        except Exception as e:
+            print(e)
+            return None
+
+    def get_block(self,block):
+        return w3.eth.get_block(block)
+    
+
+class FixedRate:
```

Comment:
it should take another param:  web3 connection, so we can use it in the functions below